shared/ssz/README.md

```diff
@@ -23,6 +23,14 @@ type Decodable interface {
 }
 ```
 
+### Hashable
+A type is Hashable if it implements `HashSSZ()`.
```

Comment:
Can we rename `HashSSZ` to `TreeHashSSZ` to be more explicit? since there's still a more general hashing function that could be used